tests/flows/block_hash_publish_receive.go

```diff
@@ -29,19 +31,23 @@ func BlockHashPublishReceive(network interfaces.Network) {
 		subnetAInfo.BlockchainID,
 	)
 
-	// gather expectations
-
-	expectedBlockNumberU64, err := subnetAInfo.RPCClient.BlockNumber(ctx)
+	// coreth and subnet-evm have different Block implementations,
+	// which means that the block hashes will be different when queried from using different clients
+	// To workaround this, we need to query the block hash from the coreth client
+	// TODO: Design a unified interface that accounts for this different
+	rpcUri := utils.HttpToRPCURI(subnetAInfo.NodeURIs[1], "C")
```

Comment:
can we set a constant for c chain blockchain ID

Reply:
any particular reason we use index 1 instead of 0?

Reply:
It shouldn't really matter, but we use 1 here since `subnetAInfo.RPCClient` already uses the URI at index 0.